WooCommerce/Classes/ViewRelated/Orders/Order Creation/OrderStatusSection.swift

```diff
@@ -0,0 +1,88 @@
+import SwiftUI
+import Yosemite
+
+struct OrderStatusSection: View {
+    let geometry: GeometryProxy
+    let dateCreated: Date
+    let orderStatus: OrderStatus
+
+    var body: some View {
+        Divider()
+
+        VStack(alignment: .leading) {
+            Text(dateString)
+                .footnoteStyle()
+
+            HStack {
+                Text(statusBadgeTitle)
+                    .foregroundColor(.black)
+                    .footnoteStyle()
+                    .padding(.horizontal, Layout.StatusBadge.horizontalPadding)
+                    .padding(.vertical, Layout.StatusBadge.verticalPadding)
+                    .background(Color(statusBadgeColor))
+                    .cornerRadius(Layout.StatusBadge.cornerRadius)
+                Spacer()
+                Button(Localization.editButton) {}
+                    .buttonStyle(LinkButtonStyle())
+                    .fixedSize(horizontal: true, vertical: true)
+                    .padding(.trailing, -Layout.linkButtonTrailingPadding) // remove trailing padding to align button title to the side
+            }
+        }
+        .padding(.horizontal, insets: geometry.safeAreaInsets)
+        .padding([.leading, .trailing, .top])
+        .background(Color(.listForeground))
+
+        Divider()
+    }
+
+    private var dateString: String {
+        let formatter = DateFormatter.mediumLengthLocalizedDateFormatter
+
+        return formatter.string(from: dateCreated)
+    }
+
+    private var statusBadgeTitle: String {
+        orderStatus.name ?? orderStatus.slug
+    }
```

Comment:
This could(should?) come from the view model, also it would make it more testable!